src/collections/domain/models/CollectionSearchCriteria.ts

```diff
@@ -1,16 +1,38 @@
 import { CollectionItemType } from './CollectionItemType'
+import { FilterQuery, OrderType, SortType } from './GetCollectionItemsQueryParams'
 
 export class CollectionSearchCriteria {
   constructor(
     public readonly searchText?: string,
-    public readonly itemTypes?: CollectionItemType[]
+    public readonly itemTypes?: CollectionItemType[],
+    public readonly sort?: SortType,
+    public readonly order?: OrderType,
+    public readonly filterQueries?: FilterQuery[]
   ) {}
 
   withSearchText(searchText: string | undefined): CollectionSearchCriteria {
-    return new CollectionSearchCriteria(searchText, this.itemTypes)
+    return new CollectionSearchCriteria(searchText, this.itemTypes, this.sort, this.order)
   }
 
   withItemTypes(itemTypes: CollectionItemType[] | undefined): CollectionSearchCriteria {
-    return new CollectionSearchCriteria(this.searchText, itemTypes)
+    return new CollectionSearchCriteria(this.searchText, itemTypes, this.sort, this.order)
```

Comment:
To be consistent it looks like think these calls need a filterQueries parameter
```suggestion
    return new CollectionSearchCriteria(this.searchText, itemTypes, this.sort, this.order, this.filterQueries)
```